### ISSUE TITLE
Variable ($__interval_ms) not replaced in Alert Query using SQL DataSource

### ISSUE THREAD

**Issue by Joe2099** [grafana/grafana#11555]
Running 5.0.4 against MySQL datasource. Errors logged is:
lvl=eror msg="Alert Rule Result Error" logger=alerting.evalHandler ruleId=0 name="Backup alert" error="tsdb.HandleRequest() response error &{Error 1054: Unknown column '$__interval_ms' in 'group statement'  A 0xc4207e9b10 [] []}" changing state to=alerting

Actual query sent to MySQL:
```
SELECT
  UNIX_TIMESTAMP(Updated) as time_sec,
  Avg(Value) as value,
  Name as metric
FROM measurements
INNER JOIN sensors ON sensors.SID = measurements.SID 
WHERE Updated >= FROM_UNIXTIME(1523411324) AND Updated <= FROM_UNIXTIME(1523414924) AND measurements.SID = 190
GROUP BY UNIX_TIMESTAMP(Updated) DIV ($__interval_ms/1000) 
ORDER BY Updated ASC
```

Notice the Grafana $__interval_ms is still in the query causing it to fail.

Expected results was the $__interval_ms would be replaced with a value such that the MySQL is valid.

**Comment by Joe2099** [comment]
I believe #11399 and #11555 got mixed up in the previous note.  
#11399 refers to **fix** #10555 and not **issue** #10555.

I do believe this is a bug and not a feature request as the functionality (alerting) does not work as intended.

**Comment by hluaces** [comment]
Same thing happening here. 

I've got a few template variables used on several queries and all of them fail because they are not being properly expanded.

Dummy query (not the actual one, just an example):
```
SELECT 		
  unix_timestamp(now()) AS time_sec, 
  t.status AS metric, 
  count(*) AS value
FROM
    departments t 
WHERE 
    t.name IN ($departamentos)
```

Result:
```
firing:true
state:"alerting"
conditionEvals:" = true"
timeMs:"12.362ms"
error:"tsdb.HandleRequest() response error &{Error 1054: Unknown column '$departamentos' in 'where clause' A 0xc42a993680 [] []}"
```

**Comment by marefr** [comment]
Template variables is currently not supported when using alerting! See #6557

**Comment by hluaces** [comment]
Thanks for pointing me to the issue. I'll be sure to follow it. I probably misunderstood this one.

Thank you very much.

**Comment by Joe2099** [comment]
Issue is still present in 5.2.1.
t=2018-07-19T20:08:01-0500 lvl=eror msg="Alert Rule Result Error" logger=alerting.evalContext ruleId=1 name="Test alert" error="tsdb.HandleRequest() response error &{Error 1054: Unknown column '$__interval_ms' in 'group statement'  A 0xc420646350 [] []}" changing state to=keep_state

**Comment by saurabhnanda** [comment]
Is there any workaround for this at all? I'd like to use the nifty variable selector in the UI to switch between different hosts. I am fine with defining alerts per panel+host combination (due to the current limitation the alerts engine has wrt variables/templates)

**Comment by marefr** [comment]
@Joe2099 @saurabhnanda this was fixed and released in Grafana v5.3

**Comment by saurabhnanda** [comment]
@marefr I think I commented on the wrong issue. I can confirm that `$__interval` macro works fine with alerts. I was referring to the ability of having custom variables on a dashboard, along with alerts that are define _without_ using the variable (due to this limitation - https://github.com/grafana/grafana/issues/6557 )

Currently, I was forced to define two separate dashboards:

- Per-host dashboard without alerts
  - custom_variable=hostname
  - panels = cpu usage, disk usage, network transfers, etc. 
- Combined dashboard solely for alerting
  - no custom variable
  - panels = cpu usage split by hosts (i.e. graph has multiple series), similarly for other panels
  - alerts which fire whenever _any_ host crosses cpu, memory, or disk threshold.
